packages/peregrine/lib/context/cart.js

```diff
@@ -72,3 +97,31 @@ export default connect(
 )(CartContextProvider);
 
 export const useCartContext = () => useContext(CartContext);
+
+export const CartContextResolvers = {
+    Query: {
+        cartId: async (root, args, context) => {
+            try {
+                console.log('fetching cart id in resolver');
+                let cartId;
+                try {
+                    // Will throw if no data. Prevent this by defining initial value.
+                    const cacheData = context.cache.readQuery({
+                        query: GET_LOCAL_CART_ID
+                    });
+                    cartId = cacheData;
+                } catch (err) {
+                    console.log('No cached cart id, fetching from server');
+                    const fetchedData = await context.client.mutate({
+                        mutation: createCartMutation
+                    });
+                    cartId = fetchedData.data.cartId;
```

Comment:
This handles the creation of a cart id if one doesn't exist at all. When the app loads for the first time the cache is an empty object so the read fails and we end up here, which write the value to the cache.